File: server/src/color.cpp
//
// Created by danpashin on 8/27/22.
//
#include "color.h"

using namespace smart_led;

Color Color::morning() {
    return Color{
            .rgb = RGBColor{
                    .red = 150,
                    .green = 35,
                    .blue = 2
            },
            .white = 0,
    };
}

Color Color::daylight() {
    return Color{
            .rgb = RGBColor{
                    .red = 255,
                    .green = 80,
                    .blue = 7
            },
            .white = 255,
    };
}

Color Color::evening() {
    return Color{
            .rgb = RGBColor{
                    .red = 60,
                    .green = 20,
                    .blue = 1
            },
            .white = 0,
    };
}

Color Color::night() {
    return Color{
            .rgb = RGBColor{
                    .red = 0,
                    .green = 0,
                    .blue = 0
            },
            .white = 0,
    };
}

File: server/src/arduino/color.rs
#[repr(C)]
#[derive(Debug, Default, Copy, Clone)]
pub(crate) struct Color {
    pub(crate) red: u8,
    pub(crate) green: u8,
    pub(crate) blue: u8,
    pub(crate) white: u8,
}

impl Color {
    pub(crate) fn new(red: u8, green: u8, blue: u8, white: u8) -> Self {
        Self {
            red,
            green,
            blue,
            white,
        }
    }

    pub(crate) fn copy(self, buffer: &mut [u8]) -> u8 {
        buffer[0] = self.red;
        buffer[1] = self.green;
        buffer[2] = self.blue;
        buffer[3] = self.white;

        4
    }
}
